app/components/UserAvatarAddress.tsx

```diff
@@ -0,0 +1,82 @@
+'use client';
+
+import { ChainId, RPC_LIST } from '@/constants';
+import { shortenAddress } from '@/utils';
+import { cx } from 'class-variance-authority';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useMemo } from 'react';
+import { Address, getAddress, http } from 'viem';
+import { mainnet } from 'viem/chains';
+import { createConfig, fallback, useEnsAvatar, useEnsName } from 'wagmi';
+
+interface UserAvatarAddressProps {
+  address: Address;
+  isAIAgent: boolean;
+}
+
+const mainnetConfigForENS = createConfig({
+  chains: [mainnet],
+  transports: { [mainnet.id]: fallback([http(RPC_LIST[ChainId.MAINNET]), http()]) },
+});
+
+export const UserAvatarAddress = ({ address, isAIAgent }: UserAvatarAddressProps) => {
+  const normalizedAddress = useMemo(() => {
+    try {
+      return getAddress(address);
+    } catch (error) {
+      console.error('Invalid Ethereum address:', error);
+      return address;
+    }
+  }, [address]);
+
+  const { data: ensName } = useEnsName({
+    address: normalizedAddress,
+    chainId: mainnet.id,
+    config: mainnetConfigForENS,
+  });
+
+  const { data: ensAvatar } = useEnsAvatar({
+    name: ensName ?? undefined,
+    chainId: mainnet.id,
+    config: mainnetConfigForENS,
+  });
+
+  console.log('ensAvatar:', ensAvatar);
+
+  const GRADIENT_BASED_ON_ADDRESS_URL = 'https://avatars.jakerunzer.com';
+  const ENS_METADATA_AVATAR_URL = 'https://metadata.ens.domains/mainnet/avatar';
+
+  const avatarGradientUrl = (address: string) =>
+    `${GRADIENT_BASED_ON_ADDRESS_URL}/${address}`;
+
+  const avatarEnsUrl = (ens: string) => `${ENS_METADATA_AVATAR_URL}/${ens}`;
+
+  const avatarUrl =
+    ensName && ensAvatar ? avatarEnsUrl(ensName) : avatarGradientUrl(normalizedAddress);
```

Comment:
Should we have a loading for ensName and ensAvatar?

Reply:
I think it's okay since they already have a default  "bg gray", so it sense that isLoading